src/utils/NumberUtils.h

```diff
@@ -4,6 +4,22 @@
 #ifndef INCLUDED_NUMBERUTILS_H
 #define INCLUDED_NUMBERUTILS_H
 
+// With C++17, we can use <charconv> from_chars.
+//
+// That has advantage of not dealing with locale / errno,
+// which might involve locks or thread local storage accesses.
+//
+// Note that it is not enough to check __cplusplus version,
+// since even if compiler reports C++17 it might not implement
+// from_chars for floats. Checking __cpp_lib_to_chars works
+// correctly and the check starts passing with gcc 11, clang 12,
+// msvc 2019 (16.4). It correctly does not pass on apple clang 15,
+// since it does not implement from_chars for floats.
+#if __cpp_lib_to_chars >= 201611L
```

Comment:
Minor but to make sure:
Seems like __cpp_lib_to_chars "feature-test macros" are standardized in C++20, If I understand correctly -in theory- a C++17 compiler may not have this macro defined even if from_chars are implemented, thus we may have a false-negative here. Is that correct? 

The worst case it'll fall back to slow path, so that's totally fine but I'm just curious.

Reply:
In theory, yes. In practice, this was the "least bad" way I found that would work on all of MSVC/clang/gcc, while also working on Apple Clang (in a sense that Apple Clang does not pass the test; from_chars on floats is still not there). In theory there could be some exotic compiler that might have from_chars for floats but not expose `__cpp_lib_to_chars`.